core/src/main/scala/kafka/server/AbstractFetcherManager.scala

```diff
@@ -98,7 +100,7 @@ abstract class AbstractFetcherManager(protected val name: String, clientId: Stri
   def removeFetcherForPartitions(partitions: Set[TopicAndPartition]) {
     mapLock synchronized {
       for ((key, fetcher) <- fetcherThreadMap) {
-        fetcher.removePartitions(partitions)
+        fetcher.removePartitions(partitions.map { case TopicAndPartition(t, p) => new TopicPartition(t, p) })
```

Comment:
Ditto here. It actually looks like it saves an unnecessary conversion to `TopicAndPartition`.